### ISSUE TITLE
chisel client - Avoiding plain-text password when http_proxy

### ISSUE THREAD

**Issue by yaakov-berkovitch** [jpillora/chisel#340]
Hello,

This is not a real issue or bug, but a question, even if I think it violates some basic security rules.

We are using chisel v1.7.6.
I'm wondering if there is a way to avoid passing a plain-text password in the chisel client command line. After starting the chisel client, anyone that list the running processes will see the command arguments and so the clear password.

I didn't find a way to hide this command or the password from users, so any help with this concern will be very helpful.

Thanks.

**Comment by jpillora** [comment]
Good point, at this stage, no way to hide it from the process args (`ps aux` will display it)

Easiest change to fix this is to use an environment variable fallback for the `--proxy` option, like `CHISEL_HTTP_PROXY` and then define this env var in a systemd unit file (or similar). Feel free to raise a PR, will merge it

**Comment by yaakov-berkovitch** [comment]
Thanks @jpillora, your workaround looks working and the password is hidden.

**Comment by yaakov-berkovitch** [comment]
@tman204 your solution is also a good approach, and by reading the password from stdin will indeed make it invisible using ps.
I didn't find a PR for this change, why not creating a PR and having your changes part of releases ? 
@jpillora your advise would be very appreciated.

**Comment by ayelet-ack** [comment]
@jpillora added a PR to fix this issue.
Can you please merge it? Thanks!

**Comment by ayelet-ack** [comment]
@jpillora I opened a PR with a fix for this issue a while ago: https://github.com/jpillora/chisel/pull/361
could you please merge?
